### ISSUE TITLE
"remove" event triggered prematurely

### ISSUE THREAD

**Issue by dhritzkiv** [AmpersandJS/ampersand-view#139]
Simple test code ([also on RequireBin](http://requirebin.com/?gist=39472f56f24c18f6b6b9)):

```javascript
var View = require("ampersand-view");

var CustomView = View.extend({
  initialize: function() {
    this.on("remove", function() {
    	console.log('remove was triggered');
    });
  }
});

var view = new CustomView();
// "remove was triggered"
```

It seems that adding an event listener before render(?) for the `remove` event causes that event to trigger on initalization/render.

**Comment by cdaringe** [comment]
ill look into this

**Comment by kamilogorek** [comment]
Issue here is that we're triggering either `remove` or `render` while accessing `rendered` property.

```js
        rendered: {
            deps: ['_rendered'],
            fn: function() {
                if (this._rendered) {
                    this.trigger('render', this);
                    return true;
                }
                this.trigger('remove', this);
                return false;
            }
        }
```

And exactly this property is accessed in constructor _before render method_:

```js
function View(attrs) {
    (...)
    this._rendered = this.rendered; // prep `rendered` derived cache immediately
    if (this.autoRender && this.template) {
        this.render();
    }
}
```

**Comment by dhritzkiv** [comment]
I see.

Would decoupling the `remove` event from the `_rendered` property, and triggering the `remove` event from the `remove` method make sense? In which case, you wouldn't need both the `_rendered` property and the derived `rendered` property.

**Comment by kamilogorek** [comment]
@dhritzkiv I see that @cdaringe already took care of it, thanks! :)

**Comment by cdaringe** [comment]
@dhritzkiv, we chose the strategy specifically so the user wouldn't have to manage triggering events on their own in their over-ridden render/remove functions.  the first PR implementation did something close to exactly that, just manually triggering remove/render events in their corresponding functions.  there was some discussion, and we decided to handle it for the user.

anyway, it's a pretty simple fix!  take a peek, let me know your thoughts
